crawl.py

```diff
@@ -0,0 +1,75 @@
+from datetime import datetime, timedelta
+from os import makedirs, path
+
+import pandas as pd
+import requests
+
+import reddit_auth
+
+TOKEN = reddit_auth.get_token()
+
+def fetch_posts(number:int = 3, page:str = None):
+  """fetches # of posts, starting at page reference"""
+
+  params = {'limit': number}
+  if page:
+    params['after'] = page
+
+  headers = {
+    'User-Agent': 'ProductsFromReddit/0.0.1',
+    'Authorization': f'bearer {TOKEN}'
+  }
+  res = requests.get(
+    'https://oauth.reddit.com/r/hardwareswap/new',
+    headers = headers,
+    params = params
+  )
+
+  # Dump response data posts into dataframe for manipulation
+  posts = pd.json_normalize(res.json()['data']['children'])
+  posts.columns = posts.columns.str.replace('data.', '', regex=True)
+
+  next_page = res.json()['data']['after']
+
+  return posts, next_page
+
+
+
+def save_posts_to_json(posts):
+  "writes posts to separate json files in data/ directory"
+  
+  if not path.exists("data"):
+    makedirs("data")
+
+  for index, row in posts.iterrows():
+    post_id = row['id']
+    with open(f'data/{post_id}.json', "w") as outfile:
+      outfile.write(row.to_json())
+
+
+
+def full_crawl(start_time: datetime):
+  "fetches all posts within specified duration"
```

Comment:
[Docstring nit] Needs triple quote and slight formatting tweak

```suggestion
  """Fetches all posts within specified duration, saving posts to disk"""
```